### PR TITLE
Swift: make imported and exported modules a set

### PR DESCRIPTION
**Pull request by redsun82**
This PR introduces the ability to specify a property as an unordered `set` in `schema.py`. `set` properties:
* in QL only have the indefinite `getAFoo` getter, together with `getNumberOfFoos`
* in the dbscheme are modeled as simple `(id, prop)` tuples without `keyset` constraints
* in C++ still use `std::vector` as input (no need to pay the `std::set` or `std::unordered_set` overhead there)
* are not allowed to be children, as order is important for AST children

This is used to make imported and exported modules unordered and unindexed, as this was actually causing DB inconsistencies when multiple sources importing different modules were used to implement a merged module.

Commit by commit review is encouraged.

### CODE REVIEW

File: misc/codegen/test/test_cppgen.py
Comment (redsun82) [inline misc/codegen/test/test_cppgen.py:1]: sorry about the distracting formatting changes, I fixed the autopep hook I was relying on later in this PR

**Comment by redsun82** [comment]
> I think this LGTM, but I'd like to see a DCA run before we merge this, please.

Can I ask what do you expect to learn from a DCA run?

As far as I know, the properties I've changed are not used anywhere in the library for the moment. The extractor non-generated code did not change either. Emitted traps did but in a 1:1 correspondence.

I'm happy to trigger a run, but as language-specific CLI builds for Swift DCA are still broken and I feel the risk of codeql/semmle-code inconsistencies on their `main`s for any one of the language is high (I'm experiencing it in other PRs where I would like DCA results on), I'd prefer not to enter into a spiral of DCA retriggering if there is not much value to gain from it.

**Comment by MathiasVP** [comment]
> > I think this LGTM, but I'd like to see a DCA run before we merge this, please.
> 
> Can I ask what do you expect to learn from a DCA run?
> As far as I know, the properties I've changed are not used anywhere in the library for the moment. The extractor non-generated code did not change either. Emitted traps did but in a 1:1 correspondence.

For one thing, I'd like us to keep the habit of running DCA on large PRs (such as this one). Especially when it changes the dbscheme. Who knows how that will affect how all queries are compiled and evaluated (this may in part be nullified by the fact that we still don't use a proper stats file).

> I'm happy to trigger a run, but as language-specific CLI builds for Swift DCA are still broken and I feel the risk of codeql/semmle-code inconsistencies on their `main`s for any one of the language is high (I'm experiencing it in other PRs where I would like DCA results on), I'd prefer not to enter into a spiral of DCA retriggering if there is not much value to gain from it.

Indeed, the DCA wobbliness for Swift is annoying. I hope this will be improved in the future 🤞.

**Review by MathiasVP — DISMISSED** [review]
Also: This LGTM once DCA comes back happy 🙂.

**Review by MathiasVP — APPROVED** [review]

**Comment by redsun82** [comment]
As far as I can tell the DCA experiment was ok. There were some failures related to the github actions incident, and the alert diff which is due to some unrelated wobbliness.

**Comment by MathiasVP** [comment]
> As far as I can tell the DCA experiment was ok. There were some failures related to the github actions incident, and the alert diff which is due to some unrelated wobbliness.

Thanks for doing the experiment. I agree this looks good 👍

**Pull request merged**